cmd/internal/cli/config_fakeroot_linux.go

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package cli
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"github.com/sylabs/singularity/docs"
+	"github.com/sylabs/singularity/internal/pkg/fakeroot"
+	"github.com/sylabs/singularity/internal/pkg/sylog"
+	"github.com/sylabs/singularity/pkg/cmdline"
+)
+
+// -a|--add
+var fakerootConfigAdd bool
+var fakerootConfigAddFlag = cmdline.Flag{
+	ID:           "fakerootConfigAddFlag",
+	Value:        &fakerootConfigAdd,
+	DefaultValue: false,
+	Name:         "add",
+	ShortHand:    "a",
+	Usage:        "add a fakeroot mapping entry for a user allowing him to use the fakeroot feature",
+}
+
+// -r|--remove
+var fakerootConfigRemove bool
+var fakerootConfigRemoveFlag = cmdline.Flag{
+	ID:           "fakerootConfigRemoveFlag",
+	Value:        &fakerootConfigRemove,
+	DefaultValue: false,
+	Name:         "remove",
+	ShortHand:    "r",
+	Usage:        "remove the user fakeroot mapping entry preventing him to use the fakeroot feature",
+}
+
+// -e|--enable
+var fakerootConfigEnable bool
+var fakerootConfigEnableFlag = cmdline.Flag{
+	ID:           "fakerootConfigEnableFlag",
+	Value:        &fakerootConfigEnable,
+	DefaultValue: false,
+	Name:         "enable",
+	ShortHand:    "e",
+	Usage:        "enable a user fakeroot mapping entry allowing him to use the fakeroot feature",
+}
+
+// -d|--disable
+var fakerootConfigDisable bool
+var fakerootConfigDisableFlag = cmdline.Flag{
+	ID:           "fakerootConfigDisableFlag",
+	Value:        &fakerootConfigDisable,
+	DefaultValue: false,
+	Name:         "disable",
+	ShortHand:    "d",
+	Usage:        "disable a user fakeroot mapping entry preventing him to use the fakeroot feature",
+}
+
+// configFakerootCmd singularity config fakeroot
+var configFakerootCmd = &cobra.Command{
+	Args:                  cobra.ExactArgs(1),
+	DisableFlagsInUseLine: true,
+	PreRun:                func(cmd *cobra.Command, args []string) { EnsureRootPriv(cmd, nil) },
+	RunE: func(cmd *cobra.Command, args []string) error {
+		username := args[0]
+
+		subUIDConfig, err := fakeroot.GetConfig(fakeroot.SubUIDFile, true, nil)
```

Comment:
Maybe add `defer  subUIDConfig.Close()`? That will require to change fatals to smth different. 
Not sure what is better here.

Reply:
`Close` behind the scene flushes any pending update to the underlying `/etc/subXid` file (like after a `AddUser` call), so if we defer it and the call to `subGIDConfig.AddUser` fail for some reasons we will update `/etc/subuid` and maybe `/etc/subgid` with a wrong entry. No defer here, this is deliberate